.github/workflows/push-pr-images.yml

```diff
@@ -0,0 +1,109 @@
+name: Push PR Images
+
+on:
+  workflow_run:
+    workflows:
+    - Build
+    types:
+      - completed
+
+jobs:
+  push:
+    name: Push PR images to GHCR
+    if: |
+      github.event.workflow_run.conclusion == 'success' &&
+      github.event.workflow_run.event == 'pull_request'
+    runs-on: ubuntu-24.04
+    permissions:
+      contents: read
+      packages: write
+      statuses: write
+
+    env:
+      GHCR_REGISTRY: ghcr.io
+      GHCR_IMAGE_REPO: ${{ format('{0}/{1}', 'ghcr.io', github.repository) }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v5
+
+      - name: Download claudio assets
+        uses: actions/download-artifact@v4
+        with:
+          run-id: ${{ github.event.workflow_run.id }}
+          github-token: ${{ github.token }}
+          name: claudio-arm64-and-amd64
+        continue-on-error: true
+
+      - name: Check if artifacts are present
+        id: artifact-check
+        run: |
+          if [ -d "claudio-amd64" ] && [ -d "claudio-arm64" ]; then
```

Comment:
Minor question, if all next steps are checking the found env value.... can you make this step to exit 1 ..that should stop the flow with error no, so no other step should be executed?

Thinking out loud....

beyond this LGTM